test/drivers/test_suites/syscall_exit_suite/recvmsg_x.cpp

```diff
@@ -518,7 +533,134 @@ TEST(SyscallExit, recvmsgX_fail)
+	int64_t received_bytes, sent_bytes, msg_controllen;
+	struct msghdr msg;
+	struct cmsghdr *cmsg;
+	char cmsg_buf[CMSG_SPACE(sizeof(int))];
+	struct iovec iov = {
+		.iov_base = (void *)FULL_MESSAGE, 
+		.iov_len = FULL_MESSAGE_LEN,
+	};
+
+	/* We don't want to get any info about the connected socket so `addr` and `addrlen` are NULL. */
+	int connected_socket_fd = syscall(__NR_accept, server_socket_fd, NULL, NULL);
+	assert_syscall_state(SYSCALL_SUCCESS, "accept (server)", connected_socket_fd, NOT_EQUAL, -1);
+
+	/* Now we can fork. We still maintain the connected_socket_fd in both parent and child processes */
+	pid_t pid = fork();
+	if(pid)
+	{
+		/* Create a socket. It is used to pass it to the child process, just for test purposes */
+		int sock = socket(AF_UNIX, SOCK_STREAM, 0);
+		msg = {
+			.msg_iov = &iov,
+			.msg_iovlen = 1,
+			.msg_control = cmsg_buf,
+			.msg_controllen = sizeof(cmsg_buf)
+		};
+		msg_controllen = msg.msg_controllen;
+
+		cmsg = CMSG_FIRSTHDR(&msg);
+		cmsg->cmsg_level = SOL_SOCKET;
+		cmsg->cmsg_type = SCM_RIGHTS;
+		cmsg->cmsg_len = CMSG_LEN(sizeof(sock));
+
+		memcpy(CMSG_DATA(cmsg), &sock, sizeof(sock));
+
+		sent_bytes = syscall(__NR_sendmsg, client_socket_fd, &msg, 0);
+		assert_syscall_state(SYSCALL_SUCCESS, "sendmsg (client)", sent_bytes, NOT_EQUAL, -1);;
+
+		int wstatus;
+		waitpid(-1, &wstatus, 0);
+
+		syscall(__NR_shutdown, sock);
+		syscall(__NR_close, sock);
+	} else
+	{
+		char buf[FULL_MESSAGE_LEN];
+		struct iovec iov = {
+			iov.iov_base = (void *)buf,
+			iov.iov_len = sizeof(buf)
+		};
+
+		struct msghdr msg = {
+			.msg_iov = &iov,
+			.msg_iovlen = 1,
+			.msg_control = cmsg_buf,
+			.msg_controllen = sizeof(cmsg_buf)
+		};
+
+		msg.msg_iov = &iov;
+		msg.msg_iovlen = 1;
+		iov.iov_base = (void *)buf;
+		iov.iov_len = sizeof(buf);
+
+		received_bytes = syscall(__NR_recvmsg, connected_socket_fd, &msg, 0);
+		assert_syscall_state(SYSCALL_SUCCESS, "recvmsg (server)", received_bytes, NOT_EQUAL, -1);
+		exit(0);
+	}
+
+	/* Cleaning phase */
+	syscall(__NR_shutdown, connected_socket_fd, 2);
+	syscall(__NR_shutdown, server_socket_fd, 2);
+	syscall(__NR_shutdown, client_socket_fd, 2);
+	syscall(__NR_close, connected_socket_fd);
+	syscall(__NR_close, server_socket_fd);
+	syscall(__NR_close, client_socket_fd);
+	syscall(__NR_unlinkat, 0, UNIX_CLIENT, 0);
+	syscall(__NR_unlinkat, 0, UNIX_SERVER, 0);
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	evt_test->disable_capture();
+
+	evt_test->assert_event_presence(pid);
+
+	if(HasFatalFailure())
+	{
+		return;
+	}
+
+	evt_test->parse_event();
+
+	evt_test->assert_header();
+	
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	/* Parameter 1: res (type: PT_ERRNO) */
+	evt_test->assert_numeric_param(1, (int64_t)FULL_MESSAGE_LEN);
+
+	/* Parameter 2: size (type: PT_UINT32) */
+	evt_test->assert_numeric_param(2, (uint32_t)FULL_MESSAGE_LEN);
+
```

Comment:
I would add the check for parameter 4 here. Something like this:
```
evt_test->assert_tuple_unix_param(4, PPM_AF_UNIX, UNIX_SERVER);
```
The problem is that this only works with the modern bpf probe at the moment. I would be curious to solve this for the other two drivers. I think this is part of a bigger discussion, but the way we are constructing the tuple for UNIX sockets it's kind of wrong in my opinion: https://github.com/falcosecurity/libs/blob/a884db5834ce8e918e2733d6f0e88c1d84914a0c/driver/modern_bpf/helpers/store/auxmap_store_params.h#L665 as you can see, we are pushing two kernel pointers which are meaningless in userspace IMO. Plus, looks like the modern probe and the old drivers push a different tuple in this particular case.